src/com/shashi/service/impl/ProductServiceImpl.java

```diff
@@ -595,6 +661,34 @@ public String updateProductWithoutImage(String prevProductId, ProductBean update
 			ps.setInt(5, updatedProduct.getProdQuantity());
 			ps.setString(6, prevProductId);
 
+
+			// Check if the product has a discount
+			Discount discount = updatedProduct.getDiscount();
+			if (discount != null) {
+
+				// Update or insert the discount in the database
+				// Check if the discount already exists for the product
+				if (discountExists(discount.getProductId())) {
+
+					// Update the existing discount
+					String updateQuery = "UPDATE discount SET discountedPrice=?, startDate=?, endDate=? WHERE productId=?";
```

Comment:
updating discounts should be in the DiscountService class. if you add a DiscountId string to ProductBean, you can just do ps.setString(7, updatedDiscountId)